ch_tools/chadmin/cli/data_store_group.py

```diff
@@ -405,14 +379,42 @@ def find_paths_to_part_with_lost_keys(ctx: Context, root_path: str) -> List[str]
                 s3_client, disk_conf.bucket_name, full_key
             ):
                 logging.debug("Not found key {}", full_key)
-                result.append(path)
+                parts_paths_with_lost_keys.append(path)
+
+                partition_list = get_partitions_by_path(ctx, [path])
+                try_restore_data(ctx, partition_list, reattach, detach)
```

Comment:
check that we already handle the partition is needed. Because it doesn't make sense to run reattach/detach several time for the single partition.

Reply:
Update